images/installer/Dockerfile.ci

```diff
@@ -1,10 +1,14 @@
 # This Dockerfile is used by CI to publish the installer image.
 # It builds an image containing only the openshift-install.
 
+#FROM registry.ci.openshift.org/ocp/4.15:installer-terraform-providers as providers
+FROM quay.io/rdossant/terraform-providers:latest as providers
+
 FROM registry.ci.openshift.org/ocp/builder:rhel-8-golang-1.20-openshift-4.15 AS builder
 ARG TAGS=""
 WORKDIR /go/src/github.com/openshift/installer
 COPY . .
+COPY --from=providers /go/src/github.com/openshift/installer/terraform/bin/ terraform/bin/
```

Comment:
This LGTM, but we will need to get the terraform providers image into ART/OSBS before we can merge these changes. 

So I think we need to merge https://github.com/openshift/release/pull/39563 first. Then get the delivery repos setup in ART, then we can merge this.

We could potentially merge the non-Dockerfile parts of this PR before getting the ART delivery repos set up.

Reply:
Since the installer-terraform-providers has been already successfully pushed in ocp:4.15 (`sha256:63f7ca...`), wouldn't be enough to add: 

```
installer-terraform-providers:
    name: "4.15"
    namespace: ocp
    tag: installer-terraform-providers
```

in the `base_images` section of [openshift-installer-master.yaml](https://github.com/openshift/release/blob/afcb45b01cd397649757cf5b9b97a5f3c9efb91e/ci-operator/config/openshift/installer/openshift-installer-master.yaml#L38) (before merging this one)?

Reply:
@andfasano it seems to be working without that. Do you know what adding that stanza does?

Reply:
https://github.com/openshift/release/pull/45964

Reply:
> @andfasano it seems to be working without that. Do you know what adding that stanza does?

I think there's a default to fetch the referenced images from the current one thanks to the FROM replacement, ie https://gcsweb-ci.apps.ci.l2s4.p1.openshiftapps.com/gcs/origin-ci-test/pr-logs/pull/openshift_installer/7688/pull-ci-openshift-installer-master-images/1727325429670350848/artifacts/build-logs/installer-amd64.log: 

```
...
2023-11-22T14:13:20.748800572Z [1/3] STEP 1/2: FROM registry.ci.openshift.org/ocp/4.15:installer-terraform-providers AS providers
2023-11-22T14:13:20.751555959Z Trying to pull registry.ci.openshift.org/ocp/4.15:installer-terraform-providers...
...
```
It's safer/easier to read anyhow to configure explicitly it in the base images